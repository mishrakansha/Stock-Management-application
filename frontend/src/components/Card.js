import React, { Component } from "react";

import DeleteIcon from "@mui/icons-material/Delete";
import InfoIcon from "@mui/icons-material/Info";
import Button from "@mui/material/Button";
import EditIcon from "@mui/icons-material/Edit";
import "./css/Cards.css";
import TextField from "@mui/material/TextField";
export default class Card extends Component {
  constructor(props) {
    super(props);
    this.state = { counter: this.props.quantity };
  }
  handleIncrement = () => {
    this.setState({ counter: this.state.counter + 1 });
  };

  handleDecrement = () => {
    this.setState({ counter: this.state.counter - 1 });
  };
  counterChanged = (event) => {
    this.setState({ counter: event.target.value });
  };
  render() {
    return (
      <div class="cards">
        <div>
          <h2>{this.props.itemname}</h2>
        </div>
        <div>
          <h4> Price:-{this.props.price}</h4>
        </div>
        <div className="incrementdecrementbutton">
          <Button
            disabled={this.state.counter < 1}
            onClick={this.handleDecrement}
          >
            -
          </Button>
          <Button style={{ padding: "0px" }}>
            <TextField
              variant="outlined"
              hiddenLabel
              id="filled-hidden-label-small"
              placeholder="search item.........."
              size="small"
              margin="0"
              style={{ "max-width": "4.5em", textAlign: "center" }}
              inputProps={{
                style: {
                  textAlign: "center",
                },
              }}
              onChange={this.counterChanged}
              value={this.state.counter}
            />
          </Button>
          <Button onClick={this.handleIncrement}>+</Button>
        </div>
        <div className="crudButtons">
          <Button
            variant="contained"
            size="small"
            startIcon={<InfoIcon fontSize="inherit" />}
          >
            Show Details
          </Button>
          <Button
            variant="contained"
            size="small"
            startIcon={<EditIcon fontSize="inherit" />}
          >
            Edit
          </Button>
          <Button
            variant="contained"
            size="small"
            startIcon={<DeleteIcon fontSize="inherit" />}
          >
            Delete
          </Button>
        </div>
      </div>
    );
  }
}
