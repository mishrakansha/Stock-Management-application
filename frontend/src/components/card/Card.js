import React, { Component } from "react";
import Button from "@mui/material/Button";
import "./Cards.css";
import { Link } from "react-router-dom";
import { deleteItem } from "./../../actions/itemsFetching";
import moment from "moment";
import { connect } from "react-redux";

// import TextField from "@mui/material/TextField";
class Card extends Component {
  constructor(props) {
    super(props);
    this.state = { delete: true };
  }
  handleDelete = () => {
    const id = this.props.id;
    this.props
      .deleteItem(id)
      .then((res) => {
        this.setState({ delete: false });
      })
      .catch((e) => {
        console.log(e);
      });
  };
  render() {
    return (
      <>
        {this.state.delete && (
          <div className="cards">
            <div className="cardHeading">
              <h4>{this.props.itemName}</h4>
            </div>
            <div className="cardBody">
              <div>By:-{this.props.manufacturingCompany}</div>
              <div className="dateContainer">
                <div className="dateTextContainer">Added On</div>
                <div className="dateIconContainer">
                  <i className="fa-solid fa-calendar-days"></i>
                  {moment(this.props.date).format(" DD/MM/YYYY")}
                </div>
              </div>
              <div className="priceQuantityContainer">
                <div>Qty : {this.props.quantity}</div>
                <div>â‚¹ : {this.props.price}</div>
              </div>
              <div className="crudButtons">
                <Link
                  className="Link"
                  to={`showdetails/${this.props.id}`}
                  state={{ id: this.props.id }}
                >
                  <Button
                    variant="contained"
                    sx={{
                      width: 35,
                      minHeight: 35,
                      minWidth: 35,
                      height: 35,
                      backgroundColor: "#323765",
                      borderRadius: 0,
                      border: "1px solid",
                      borderColor: "primary.main",
                      "& .MuiButton-startIcon": { margin: 0 },
                      ":hover": {
                        bgcolor: "#323765",
                        color: "white",
                      },
                    }}
                    size="small"
                  >
                    <i className="fa-solid fa-eye"></i>
                  </Button>
                </Link>
                <Link
                  className="Link"
                  to={`editdetails/${this.props.id}`}
                  state={{ id: this.props.id }}
                >
                  <Button
                    sx={{
                      width: 35,
                      minHeight: 35,
                      minWidth: 35,
                      height: 35,
                      backgroundColor: "#323765",
                      borderRadius: 0,
                      border: "1px solid",
                      borderColor: "primary.main",
                      "& .MuiButton-startIcon": { margin: 0 },
                      ":hover": {
                        bgcolor: "#323765",
                        color: "white",
                      },
                    }}
                    variant="contained"
                    size="small"
                  >
                    <i className="fa-solid fa-pencil"></i>
                  </Button>
                </Link>
                <Button
                  sx={{
                    width: 35,
                    minHeight: 35,
                    minWidth: 35,
                    height: 35,
                    backgroundColor: "#323765",
                    borderRadius: 0,
                    border: "1px solid",
                    borderColor: "primary.main",
                    "& .MuiButton-startIcon": { margin: 0 },
                    ":hover": {
                      bgcolor: "#323765",
                      color: "white",
                    },
                  }}
                  onClick={this.handleDelete}
                  variant="contained"
                  size="small"
                >
                  <i className="fa-solid fa-trash"></i>
                </Button>
              </div>
            </div>
          </div>
        )}
      </>
    );
  }
}
export default connect(null, { deleteItem })(Card);
