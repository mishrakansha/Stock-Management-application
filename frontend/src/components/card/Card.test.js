import React from "react";
import { shallow } from "enzyme";
import Card from "./Card";
import moment from "moment";
describe("Card", () => {
  it("Take a snapshot if do not added", () => {
    const component = shallow(<Card />);
    expect(component).toMatchSnapshot();
  });
  it("Test to render card Heading", () => {
    const props = "mobile";
    const component = shallow(<Card itemName={props} />);
    expect(component.find(".cardHeading").text()).toEqual(props);
  });
  it("Test to render Manufactured By", () => {
    const props = "vivo";
    const component = shallow(<Card manufacturingCompany={props} />);
    expect(component.find(".manufacturingCompany").text()).toEqual(
      "Manufactured By : " + props
    );
  });
  it("Test to render Date field", () => {
    const props = "2023-03-25";
    const component = shallow(<Card date={props} />);
    expect(component.find(".dateContainer").text()).toEqual(
      "Date " + moment(props).format("DD/MM/YYYY")
    );
  });
  it("Test to render quantity field", () => {
    const props = 12;
    const component = shallow(<Card quantity={props} />);
    expect(component.find(".quantityContainer").text()).toEqual(
      "Qty : " + props
    );
  });
  it("Test to render price field", () => {
    const props = 1020;
    const component = shallow(<Card price={props} />);
    expect(component.find(".priceContainer").text()).toEqual("â‚¹ : " + props);
  });
});
