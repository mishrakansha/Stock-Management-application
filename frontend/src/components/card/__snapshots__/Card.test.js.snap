// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Take a snapshot if do not added 1`] = `
<Fragment>
  <div
    className="cards"
  >
    <div
      className="cardHeading"
    >
      <h4 />
    </div>
    <div
      className="cardBody"
    >
      <div
        className="manufacturingCompany"
      >
        Manufactured By : 
      </div>
      <div
        className="dateContainer"
      >
        <div
          className="dateTextContainer"
        >
          Date
        </div>
        <div
          className="dateIconContainer"
        >
          <i
            className="fa-solid fa-calendar-days"
          />
           03/04/2023
        </div>
      </div>
      <div
        className="priceQuantityContainer"
      >
        <div
          className="quantityContainer"
        >
          Qty : 
        </div>
        <div
          className="priceContainer"
        >
          â‚¹ : 
        </div>
      </div>
      <div
        className="crudButtons"
      >
        <Link
          className="Link"
          state={
            Object {
              "id": undefined,
            }
          }
          title="Preview "
          to="showdetails/undefined"
        >
          <ForwardRef(Button)
            size="small"
            sx={
              Object {
                "& .MuiButton-startIcon": Object {
                  "margin": 0,
                },
                ":hover": Object {
                  "bgcolor": "#323765",
                  "color": "white",
                },
                "backgroundColor": "#323765",
                "border": "1px solid",
                "borderColor": "primary.main",
                "borderRadius": 0,
                "height": 35,
                "minHeight": 35,
                "minWidth": 35,
                "width": 35,
              }
            }
            variant="contained"
          >
            <i
              className="fa-solid fa-eye"
            />
          </ForwardRef(Button)>
        </Link>
        <ForwardRef(Button)
          onClick={[Function]}
          size="small"
          sx={
            Object {
              "& .MuiButton-startIcon": Object {
                "margin": 0,
              },
              ":hover": Object {
                "bgcolor": "#323765",
                "color": "white",
              },
              "backgroundColor": "#323765",
              "border": "1px solid",
              "borderColor": "primary.main",
              "borderRadius": 0,
              "height": 35,
              "minHeight": 35,
              "minWidth": 35,
              "width": 35,
            }
          }
          title="Edit "
          variant="contained"
        >
          <i
            className="fa-solid fa-pencil"
          />
        </ForwardRef(Button)>
        <ForwardRef(Button)
          onClick={[Function]}
          size="small"
          sx={
            Object {
              "& .MuiButton-startIcon": Object {
                "margin": 0,
              },
              ":hover": Object {
                "bgcolor": "#323765",
                "color": "white",
              },
              "backgroundColor": "#323765",
              "border": "1px solid",
              "borderColor": "primary.main",
              "borderRadius": 0,
              "height": 35,
              "minHeight": 35,
              "minWidth": 35,
              "width": 35,
            }
          }
          title="Delete"
          variant="contained"
        >
          <i
            className="fa-solid fa-trash"
          />
        </ForwardRef(Button)>
      </div>
    </div>
  </div>
  <ForwardRef(Dialog)
    maxWidth="sm"
    open={false}
  >
    <ForwardRef(DialogTitle)>
      Are you sure to delete this item?
    </ForwardRef(DialogTitle)>
    <ForwardRef(DialogActions)>
      <ForwardRef(Button)
        onClick={[Function]}
        sx={
          Object {
            ":hover": Object {
              "background": "#323765",
              "transform": "scale(1.01)",
            },
            "background": " #323765",
          }
        }
        variant="contained"
      >
        Cancel
      </ForwardRef(Button)>
      <ForwardRef(Button)
        onClick={[Function]}
        sx={
          Object {
            ":hover": Object {
              "background": "#323765",
              "transform": "scale(1.01)",
            },
            "background": " #323765",
          }
        }
        variant="contained"
      >
        Confirm
      </ForwardRef(Button)>
    </ForwardRef(DialogActions)>
  </ForwardRef(Dialog)>
</Fragment>
`;
